### PR TITLE
hyprland: 0.37.1 -> 0.38.1;  hyprcursor: 0.1.5 -> 0.1.6; hyprlandPlugins.hy3: 0.36.0-unstable-2024-03-06 -> 0.38.0

### PR DESCRIPTION
**Pull request by wineee**
## Description of changes
close: https://github.com/NixOS/nixpkgs/pull/300861

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by iynaix — APPROVED** [review]

**Comment by fufexan** [comment]
@wineee 0.38.1 is out, can you update?

File: pkgs/by-name/hy/hyprcursor/fix-icon-dirs.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/libhyprcursor/hyprcursor.cpp b/libhyprcursor/hyprcursor.cpp
+index 304ab9f..1f7e95d 100644
+--- a/libhyprcursor/hyprcursor.cpp
++++ b/libhyprcursor/hyprcursor.cpp
+@@ -14,7 +14,7 @@
+ using namespace Hyprcursor;
+
+ // directories for lookup
+-constexpr const std::array<const char*, 1> systemThemeDirs = {"/usr/share/icons"};
++constexpr const std::array<const char*, 2> systemThemeDirs = {"/usr/share/icons", "/run/current-system/sw/share/icons"};
```
Comment (wineee) [inline pkgs/by-name/hy/hyprcursor/fix-icon-dirs.patch:10]: I copy the patch form https://github.com/hyprwm/hyprcursor/blob/main/nix/dirs.patch
but I still get error:
```
[LOG] Created a bezier curve, baked 255 points, mem usage: 4.08kB, time to bake: 18.02µs. Estimated average calc time: 0.04µs.
terminate called after throwing an instance of 'std::filesystem::__cxx11::filesystem_error'
  what():  filesystem error: status: Permission denied [/usr/share/icons]

```

Comment (wineee) [reply pkgs/by-name/hy/hyprcursor/fix-icon-dirs.patch:10]: Also we should pay attention to the `XCURSOR_PATH` environment variable

**Comment by JohnRTitor** [comment]
Note: PR title should be updated to reflect the change to Hyprland 38.1.

**Review by aacebedo — APPROVED** [review]

File: pkgs/by-name/hy/hyprcursor/fix-icon-dirs.patch (hunk 1 above)
Comment (fufexan) [reply pkgs/by-name/hy/hyprcursor/fix-icon-dirs.patch:10]: What if you change it to remove `/usr/share/icons` completely?

I agree I should do a more extensive review of the code and try to patch it properly.

**Comment by Jackaed** [comment]
Is this still a draft or can this now be marked as ready for review?

**Comment by wineee** [comment]
LGTM now

File: pkgs/by-name/hy/hyprcursor/fix-icon-dirs.patch (hunk 1 above)
Comment (wineee) [reply pkgs/by-name/hy/hyprcursor/fix-icon-dirs.patch:10]: Works on hyprcursor 0.1.6

File: pkgs/applications/window-managers/hyprwm/hyprland/default.nix [hunk 2]
```diff
@@ -73,7 +73,7 @@ stdenv.mkDerivation (finalAttrs: {
       --replace "@HASH@" '${finalAttrs.src.rev}' \
       --replace "@BRANCH@" "" \
       --replace "@MESSAGE@" "" \
-      --replace "@DATE@" "2024-03-16" \
+      --replace "@DATE@" "2024-04-02" \
```
Comment (fufexan) [inline pkgs/applications/window-managers/hyprwm/hyprland/default.nix:76]:
```suggestion
      --replace "@DATE@" "2024-04-06" \
```

File: pkgs/by-name/hy/hyprcursor/package.nix [hunk 3]
```diff
@@ -7,19 +7,22 @@
...
 
   src = fetchFromGitHub {
     owner = "hyprwm";
     repo = "hyprcursor";
     rev = "refs/tags/v${finalAttrs.version}";
-    hash = "sha256-e6+fu30inlTIdflotS6l7qYusslKMNkhZVNLn9ZSogg=";
+    hash = "sha256-qwmR2p1oc48Bj7gUDvb1oGL19Rjs2PmEmk4ChV01A5o=";
   };
 
+  patches = [ ./fix-icon-dirs.patch ];
```
Comment (fufexan) [inline pkgs/by-name/hy/hyprcursor/package.nix:24]:
```suggestion
  patches = [
    # fix icon directories system search path
    ./fix-icon-dirs.patch
  ];
```

**Review by fufexan — CHANGES_REQUESTED** [review]

**Comment by fufexan** [comment]
Looks like hyprcursor 1.7 was just released.

File: pkgs/by-name/hy/hyprcursor/fix-icon-dirs.patch (hunk 1 above)
Comment (fufexan) [reply pkgs/by-name/hy/hyprcursor/fix-icon-dirs.patch]: Does the included patch work on 0.1.7?

**Comment by SoSeDiK** [comment]
Not necessarily targeted for this PR, but could you consider adding support for [hyprcursor-util](https://github.com/hyprwm/hyprcursor/tree/main/hyprcursor-util) to allow generation/management of themes?

File: pkgs/by-name/hy/hyprcursor/fix-icon-dirs.patch (hunk 1 above)
Comment (wineee) [reply pkgs/by-name/hy/hyprcursor/fix-icon-dirs.patch]: > Does the included patch work on 0.1.7?

Yes

**Review by fufexan — APPROVED** [review]

**Pull request merged**